src/polyswarmd/artifacts/ipfs.py

```diff
@@ -0,0 +1,256 @@
+import base58
+import functools
+import json
+import logging
+import re
+import uuid
+
+from requests import HTTPError
+
+from polyswarmd.artifacts.client import AbstractArtifactServiceClient, ArtifactServiceException
+
+logger = logging.getLogger(__name__)
+
+
+class InvalidIpfsHashException(Exception):
```

Comment:
I would recommend that exceptions for the whole package should be put in another module. Would be great if we could have some kind of `PolyswarmDException` from which all exceptions extend from. This makes us able to catch any user-generated exception.

Reply:
Will do.